pkg/handler/scale_handler.go

```diff
@@ -214,7 +220,8 @@ func (h *ScaleHandler) GetDeploymentScalers(scaledObject *kedav1alpha1.ScaledObj
 			serviceAccount := &v1.ServiceAccount{}
 			err = h.client.Get(context.TODO(), types.NamespacedName{Name: serviceAccountName, Namespace: scaledObject.GetNamespace()}, serviceAccount)
 			if err != nil {
-				return scalers, nil, fmt.Errorf("error getting deployment: %s", err)
+				closeScalers(scalersRes)
+				return []scalers.Scaler{}, nil, fmt.Errorf("error getting deployment: %s", err)
```

Comment:
Looks like the existing error message was wrong. It should probably be `error getting service account: %w` Would be nice to update it in this change as well.

Reply:
FIxed